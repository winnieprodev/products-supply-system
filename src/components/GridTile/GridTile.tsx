import React from "react";

import * as Styled from "./gridTile.styled";

interface BaseOwnProps {
  onClick: () => void;
  title: string;
  image?: string;
  icon?: React.ReactNode;
  mono?: string;
}

interface WithImageProps extends BaseOwnProps {
  image: string;
}

interface WithIconProps extends BaseOwnProps {
  icon: React.ReactNode;
}
interface WithMonoProps extends BaseOwnProps {
  mono: string;
}

type OwnProps = WithImageProps | WithMonoProps | WithIconProps;

const GridTile: React.FC<React.PropsWithChildren<OwnProps>> = ({
  onClick,
  title,
  image,
  icon,
  mono,
  children,
}) => {
  return (
    <Styled.Wrapper onClick={onClick}>
      <Styled.Tile>
        {icon && <Styled.Icon>{icon}</Styled.Icon>}
        {image && <Styled.Image src={image} />}
        {mono && <Styled.Mono>{mono}</Styled.Mono>}
        {children}
      </Styled.Tile>
      <Styled.Title>{title}</Styled.Title>
    </Styled.Wrapper>
  );
};

export default React.memo(GridTile);
