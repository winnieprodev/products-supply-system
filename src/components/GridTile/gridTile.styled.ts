import styled from "styled-components";
import {
  monoTextColor,
  secondaryColor,
  secondaryColorLighter,
} from "styles/colors";

export const Tile = styled.div`
  background: ${secondaryColorLighter};
  border-radius: 10px;
  position: relative;
  justify-content: center;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  cursor: pointer;

  &:hover ${Tile} {
    background: ${secondaryColor};
  }
`;

export const Mono = styled.div`
  color: ${monoTextColor};
  text-shadow: 0 5px 10px white;
  font-size: 6vw;
  font-weight: bold;
  text-align: center;
  line-height: calc(8vw + 80px);
`;

export const Image = styled.img`
  width: 60%;
  max-height: 8vw;
  padding: 40px;
`;

export const Icon = styled.div`
  color: ${monoTextColor};
  font-size: 6vw;
  font-weight: bold;
  text-align: center;
  line-height: 11vw;
`;

export const Title = styled.div`
  margin: 8px;
  text-align: center;
  font-size: 16px;
  font-weight: bold;
`;

export const Children = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
`;
