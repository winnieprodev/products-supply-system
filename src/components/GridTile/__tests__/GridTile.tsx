import { render } from "@testing-library/react";

import GridTile from "../GridTile";

describe("GridTile", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for image", () => {
      const { container } = render(
        <GridTile title="test" image="img.png" onClick={jest.fn()}></GridTile>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for mono", () => {
      const { container } = render(
        <GridTile title="test" mono="TES" onClick={jest.fn()}></GridTile>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
