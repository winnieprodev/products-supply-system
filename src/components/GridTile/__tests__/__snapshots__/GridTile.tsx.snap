// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridTile Snapshot tests it matches the snapshot for image 1`] = `
.c2 {
  background: #F3F3F3;
  border-radius: 10px;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
}

.c0:hover .c1 {
  background: #EEEEEE;
}

.c3 {
  width: 60%;
  max-height: 8vw;
  padding: 40px;
}

.c4 {
  margin: 8px;
  text-align: center;
  font-size: 16px;
  font-weight: bold;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <img
      class="c3"
      src="img.png"
    />
  </div>
  <div
    class="c4"
  >
    test
  </div>
</div>
`;

exports[`GridTile Snapshot tests it matches the snapshot for mono 1`] = `
.c2 {
  background: #F3F3F3;
  border-radius: 10px;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
}

.c0:hover .c1 {
  background: #EEEEEE;
}

.c3 {
  color: rgba(0,0,0,0.25);
  text-shadow: 0 5px 10px white;
  font-size: 6vw;
  font-weight: bold;
  text-align: center;
  line-height: calc(8vw + 80px);
}

.c4 {
  margin: 8px;
  text-align: center;
  font-size: 16px;
  font-weight: bold;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <div
      class="c3"
    >
      TES
    </div>
  </div>
  <div
    class="c4"
  >
    test
  </div>
</div>
`;
